include/nitro/broken_options/parser.ipp

```diff
@@ -39,68 +39,78 @@ namespace nitro
 {
 namespace broken_options
 {
-    broken_options::option& parser::option(const std::string& name, const std::string& description)
+    std::map<std::string, broken_options::option&> parser::get_all_options()
     {
-        if (multi_options_.count(name) > 0)
+        std::map<std::string, broken_options::option&> tmp;
+        for (auto& sg : groups_)
         {
-            raise<parser_error>("Trying to redefine multi_option as option. Name: ", name);
+            auto add = sg.get_all_options();
+            tmp.insert(add.begin(), add.end());
         }
+        return tmp;
+    }
 
-        if (toggles_.count(name) > 0)
+    std::map<std::string, broken_options::multi_option&> parser::get_all_multi_options()
+    {
+        std::map<std::string, broken_options::multi_option&> tmp;
+        for (auto& sg : groups_)
         {
-            raise<parser_error>("Trying to redefine toggle as multi_option. Name: ", name);
+            auto add = sg.get_all_multi_options();
+            tmp.insert(add.begin(), add.end());
         }
+        return tmp;
+    }
 
-        if (options_.count(name) == 0)
+    std::map<std::string, broken_options::toggle&> parser::get_all_toggles()
+    {
+        std::map<std::string, broken_options::toggle&> tmp;
+        for (auto& sg : groups_)
         {
-            auto res = options_.emplace(std::piecewise_construct, std::forward_as_tuple(name),
-                                        std::forward_as_tuple(name, description));
-            arguments_.add(res.first->second);
+            auto add = sg.get_all_toggles();
+            tmp.insert(add.begin(), add.end());
         }
-        return options_.at(name);
+        return tmp;
     }
 
-    broken_options::multi_option& parser::multi_option(const std::string& name,
-                                                       const std::string& description)
+    nitro::broken_options::group& parser::group(const std::string& name,
+                                                const std::string& description)
     {
-        if (options_.count(name) > 0)
+        for (nitro::broken_options::group& grp : groups_)
         {
-            raise<parser_error>("Trying to redefine option as multi_option. Name: ", name);
+            if (grp.name == name)
+                if (description.length())
+                    grp.description = description;
+            return grp;
         }
 
-        if (toggles_.count(name) > 0)
-        {
-            raise<parser_error>("Trying to redefine toggle as multi_option. Name: ", name);
-        }
+        groups_.emplace_back(group(name, description));
+        return groups_.back();
+    }
+    nitro::broken_options::group& parser::default_group(const std::string& name,
+                                                        const std::string& description)
+    {
+        if (name.length())
```

Comment:
First of all, only Java programmers use `length`, use `size` instead.
And secondly, use `!empty()` .